internal/ai/embedding.go

```
@@ -0,0 +1,311 @@
+		})
+	}
+
+	m.reindexMu.Lock()
+	defer m.reindexMu.Unlock()
+
+	tx, err := m.db.BeginTxx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		m.lo.Error("error beginning reindex transaction", "error", err)
+		return err
+	}
+	defer tx.Rollback()
```

golangci-lint (2.12.2)

[error] Error return value of `tx.Rollback` is not checked

(errcheck)